test/ec_internal_test.c

```diff
@@ -259,6 +259,96 @@ static int underflow_test(void)
 }
 #endif
 
+/*
+ * Tests behavior of the decoded_from_explicit_params flag and API
+ */
+static int decoded_flag_test(void)
+{
+    EC_GROUP *grp;
+    EC_GROUP *grp_copy = NULL;
+    ECPARAMETERS *ecparams = NULL;
+    ECPKPARAMETERS *ecpkparams = NULL;
+    EC_KEY *key = NULL;
+    unsigned char *encodedparams = NULL;
+    const unsigned char *encp;
+    int encodedlen;
+    int testresult = 0;
+
+    grp = EC_GROUP_new(EC_GFp_simple_method());
```

Comment:
Can I be annoying a little longer and propose to have a comment before each test that explicitly describes conditions and expectations?

I won't hold my approval if this is not done, it's just a suggestion.

Reply:
OK, done.